doc/Language/grammar_tutorial.pod6

```diff
@@ -567,6 +567,41 @@ Here's the final code:
     }
     =end code
 
+=head2 Add actions directly
+
+Above we see how to associate grammars with a action objects and perform
+actions on the match object. However, when we want to deal with the match
+object, that isn't the only way. See the example below:
+
+    =begin code
+    grammar G {
+      rule TOP { <function-define> }
+      rule function-define {
+        'sub' <identifier> {
+          say "func " ~ $<identifier>.made;
+          make $<identifier>.made;
+        } '(' <parameter> ')' '{' '}' {
+          say "end " ~ $/.made;
+        }
+      }
+      token identifier { \w+ { make ~$/; } }
+      token parameter { \w+ { say "param " ~ $/; } }
+    }
+
+    G.parse('sub f ( a ) { }');
+    # OUTPUT: «func f␤param a␤end f␤»
+    =end code
+
+This example is a reduced portion of a parser. Let's focus more on the feature
+it shows.
+
+First, we can add actions inside the grammar ifself, and such actions are
+performed once the match arrive them. Note that action object's method will be
```

Comment:
"once the match arrive them", might want to rewrite it as "arrives at them". I'm not sure if a match "arrives" at a certain action either.

Reply:
```
grammar G {
  rule TOP {
    | a { say "42" } b
    | a { say "12" } c
  }
}

G.parse("a c");
```
I want to show that both `say "42"` and `say "12"` are reached, as the example above.
How can I express this?

Reply:
"once their [corresponding] patterns are matched"?